libs/remote_storage/src/simulate_failures.rs

```diff
@@ -0,0 +1,123 @@
+//! This module provides a wrapper around a real RemoteStorage implementation that
+//! causes some operations to fail. For testing purposes.
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+use crate::{Download, DownloadError, RemotePath, RemoteStorage, StorageMetadata};
+
+pub struct UnreliableWrapper {
+    inner: crate::GenericRemoteStorage,
+
+    fail_first: u64,
```

Comment:
NIT: the naming is somewhat confusing to me, I always try to read it as `fail_fast` 🙂 
Maybe, `attempts_to_fail`?

Reply:
renamed